tools/generate-function-bindings/scripts/generate.ts

```diff
@@ -0,0 +1,415 @@
+// Copyright (c) HashiCorp, Inc
+// SPDX-License-Identifier: MPL-2.0
+
+// uses the functions.json file to generate the bindings for CDKTF
+
+import fs from "fs/promises";
+import * as path from "path";
+import { AttributeType } from "@cdktf/provider-generator";
+import generate from "@babel/generator";
+import template from "@babel/template";
+import * as t from "@babel/types";
+import prettier from "prettier";
+import { FUNCTIONS_METADATA_FILE } from "./fetch-metadata";
+
+const ts = template({ plugins: [["typescript", {}]] });
+
+const OUTPUT_FILE = path.resolve(
+  __dirname,
+  "..",
+  "..",
+  "..",
+  "packages",
+  "cdktf",
+  "lib",
+  "functions",
+  "terraform-functions.generated.ts"
+);
+
+type Parameter = { name: string; type: AttributeType };
+type FunctionSignature = {
+  description: string;
+  return_type: AttributeType;
+  parameters: Parameter[];
+  variadic_parameter: Parameter;
+};
+type MappedParameter = {
+  name: string;
+  mapper: string;
+  tsParam: t.Identifier;
+  docstringType: string;
+};
+
+const IMPORTS = ts`
+import {
+  anyValue,
+  asAny,
+  asBoolean,
+  asList,
+  asNumber,
+  asString,
+  listOf,
+  mapValue,
+  numericValue,
+  stringValue,
+  terraformFunction,
+  variadic,
+} from "./helpers";
+`() as t.Statement;
+t.addComment(
+  IMPORTS,
+  "leading",
+  `\n * This file is generated by tools/generate-funtion-bindings.
+ * To update this file execute 'yarn run generate-function-bindings' in the root of the repository
+ `,
+  false
+);
+
+// these are overwritten in terraform-functions.ts
+const INTERNAL_METHODS = ["join", "bcrypt", "range", "lookup"];
+
+async function fetchMetadata() {
```

Comment:
This function does create and write the program, so the function name of fetchMetadata is incorrect

Nit: fetch sounds like a network request to me, I'd prefer read.

Reply:
Oh yeah, that was a copy paste from the other script 😅